vortex-array/src/array/chunked/compute/filter.rs

```diff
@@ -88,11 +88,7 @@ fn filter_slices<'a>(
 
     for (slice_start, slice_end) in set_slices {
         let (start_chunk, start_idx) = find_chunk_idx(slice_start, chunk_ends);
-        // NOTE: we adjust slice end back by one, in case it ends on a chunk boundary, we do not
-        // want to index into the unused chunk.
-        let (end_chunk, end_idx) = find_chunk_idx(slice_end - 1, chunk_ends);
-        // Adjust back to an exclusive range
-        let end_idx = end_idx + 1;
```

Comment:
this was unnecessary, find_chunk_idx handles all the offsetting you need

Reply:
this was here b/c `find_chunk_idx` returns `(CHUNK+1, 0)` when a slice ends at the end of chunk `CHUNK`. The subtraction and subsequent addition ensures that instead this becomes `(CHUNK, chunk_len)`

Reply:
i think if you add this back, you can undo the `if end_idx != 0` below

Reply:
yes